### PR TITLE
front: add ManageTrainSchedule context and remove Pathfinding's reducer 

### LINKED ISSUE #6719: Pathfinding launched two times when adding a via
### What happened?

When adding a via via the modal or by clicking on map, the pathfinding is fired 2 times.

### What did you expect to happen?

It should be fired only once.

### How can we reproduce it (as minimally and precisely as possible)?

1. Go to a scenario
2. Add a train and launch a pathfinding
3. Open the network tab and clean the history
4. Add a via 
5. You should see two pathfinding call in the network tab

### What operating system, browser and environment are you using?

- **Browser**: Firefox v123
- **OS**: macOS Sonoma
- **Env**: Dev

### OSRD version (top right corner `Account` button > `Informations`)

dev 1482e8c

### LINKED ISSUE #9902: front: editing a train with power restrictions resets its power restrictions
### What happened?

When clicking on 'edit' for a train which has power restrictions, the power restrictions are automatically reset.

### What did you expect to happen?

The power restrictions should not be reset when clicking on editing

### How can we reproduce it (as minimally and precisely as possible)?

1. Open a scenario
2. Add a train with power restrictions (you can use a TGV)
3. See the results
4. Click on edit the train
5. See the warning displayed

### On which environments the bug occurs?

Recette (SNCF)

### On which browser the bug occurs?

Firefox

### OSRD version (top right corner `Account` button > `Informations`)

.

### LINKED ISSUE #9915: front: add Context for ManageTrainSchedule

### LINKED ISSUE #10038: No warning when removing origin or destination in the itinerary tab
### What happened?

Image

When removing the origin or destination, it should add a warning on the left of the rocket but it doesn't

We should wait for this [PR](https://github.com/OpenRailAssociation/osrd/pull/9924) to be merge before fixing this

### What did you expect to happen?

Have the warning

### How can we reproduce it (as minimally and precisely as possible)?

1. Go to a scenario
2. Add a train and launch a pathfinding
3. Remove the origin or destination

### On which environments the bug occurs?

Local

### On which browser the bug occurs?

Firefox

### OSRD version (top right corner `Account` button > `Informations`)

.

### PR DESCRIPTION
**Pull request by clarani**
closes #9915 
closes #9902 (because the power restrictions were reset by the call to usePathfinding in the RollingStockSelector)
will also close #6719
and fix #10038

### CODE REVIEW

**Comment by clarani** [comment]
Need to wait for https://github.com/OpenRailAssociation/osrd/pull/9901 first

File: front/src/modules/trainschedule/components/ManageTrainSchedule/Map.tsx [hunk 1]
```diff
@@ -385,7 +385,7 @@ const Map = ({
               layerOrder={LAYER_GROUPS_ORDER[LAYERS.LINE_SEARCH.GROUP]}
               infraID={infraID}
             />
-            <RenderPopup pathProperties={pathProperties} />
+            {!showStdcmAssets && <RenderPopup pathProperties={pathProperties} />}
```
Comment (emersion) [inline front/src/modules/trainschedule/components/ManageTrainSchedule/Map.tsx:393]: Hm, why this change? Is this fixing a bug?

Comment (clarani) [reply front/src/modules/trainschedule/components/ManageTrainSchedule/Map.tsx:393]: This popup is displayed only if I click on the map. This feature is only available in the EEx, and inside the component, I use the ManageTrainScheduleContext.
If this component renders in Stdcm, then the app crashed because there is no ManageTrainScheduleContextProvider.

File: front/src/modules/pathfinding/hooks/usePathfinding.ts [hunk 2]
```diff
@@ -228,193 +199,193 @@ export const usePathfinding = (
...
+    }
 
-    const startPathFinding = async () => {
-      setPathProperties(undefined);
-      if (pathfindingState.running) {
+    try {
+      const pathfindingResult = await postPathfindingBlocks(pathfindingInput).unwrap();
+
+      if (pathfindingResult.status === 'success') {
+        await populateStoreWithPathfinding(steps as PathStep[], pathfindingResult);
```
Comment (emersion) [inline front/src/modules/pathfinding/hooks/usePathfinding.ts]: Why do we need a cast here? Could we replace it with:

```ts
steps.map((step) => step!)
```

which should be safer in case the type of `steps` is changed to something else?

File: front/src/modules/pathfinding/hooks/usePathfinding.ts [hunk 3]
```diff
@@ -228,193 +199,193 @@ export const usePathfinding = (
...
 
-      if (!pathfindingInput) {
-        dispatch(
-          setFailure({
-            name: t('pathfinding'),
-            message: t('pathfindingMissingParamsSimple'),
-          })
+      if (incompatibleConstraintsCheck) {
+        await populateStoreWithPathfinding(
+          steps as PathStep[],
```
Comment (emersion) [inline front/src/modules/pathfinding/hooks/usePathfinding.ts]: Ditto

File: front/src/modules/pathfinding/hooks/usePathfinding.ts [hunk 4]
```diff
@@ -228,193 +199,193 @@ export const usePathfinding = (
...
-          ...(correspondingOp && {
-            name: correspondingOp.name,
-            uic: correspondingOp.uic,
-            ch: correspondingOp.ch,
-            kp: correspondingOp.kp,
-            coordinates: correspondingOp.coordinates,
-          }),
-        };
-      });
+  const launchPathfinding = async (newPathSteps?: (PathStep | null)[]) => {
```
Comment (emersion) [inline front/src/modules/pathfinding/hooks/usePathfinding.ts]: Do we need `newPathSteps` to be optional here? Doesn't seem like any caller sets it to `undefined`.

Comment (emersion) [inline front/src/modules/pathfinding/hooks/usePathfinding.ts]: Should we `useCallback` here, to avoid always returning a new function inside the context even if all dependencies haven't changed?

File: front/src/modules/pathfinding/hooks/usePathfinding.ts [hunk 5]
```diff
@@ -228,193 +199,193 @@ export const usePathfinding = (
...
-
-    if (infra && infra.state === 'CACHED' && pathfindingState.mustBeLaunched) {
-      startPathFinding();
     }
-  }, [pathfindingState.mustBeLaunched, infra]);
+  };
 
-  useEffect(() => setIsPathfindingInitialized(true), []);
+  useEffect(() => {
+    launchPathfinding();
```
Comment (emersion) [inline front/src/modules/pathfinding/hooks/usePathfinding.ts]: Previously we'd re-launch the pathfinding when the infra state switches from loading to cached. Now we don't do this anymore. Do we want to retain this behavior?

File: front/src/applications/operationalStudies/hooks/useManageTrainScheduleContext.tsx [hunk 6]
```diff
@@ -2,17 +2,26 @@ import { createContext, useContext, useMemo, type ReactNode, useState, useEffect
...
 import type { ManageTrainSchedulePathProperties } from '../types';
 
+
 type ManageTrainScheduleContextType = {
   pathProperties?: ManageTrainSchedulePathProperties;
   setPathProperties: (pathProperties?: ManageTrainSchedulePathProperties) => void;
   voltageRanges: RangedValue[];
+  launchPathfinding: (pathSteps?: (PathStep | null)[]) => void;
+  pathfindingState: PathfindingState;
+  infraInfos: { infra?: InfraWithState; reloadCount: number };
```
Comment (emersion) [inline front/src/applications/operationalStudies/hooks/useManageTrainScheduleContext.tsx]: Pedantic nit: "information" is uncountable, so shouldn't get pluralized.

**Review by emersion — COMMENTED** [review]
Excellent work! Thanks so much for splitting into separate commits. Here are a few comments :)

File: front/src/modules/trainschedule/components/ManageTrainSchedule/Map.tsx (hunk 1 above)
Comment (emersion) [reply front/src/modules/trainschedule/components/ManageTrainSchedule/Map.tsx:393]: Aha, makes sense!

File: front/src/modules/pathfinding/hooks/usePathfinding.ts (hunk 5 above)
Comment (clarani) [reply front/src/modules/pathfinding/hooks/usePathfinding.ts]: Yes you're right, we should keep this behavior

File: front/src/modules/pathfinding/components/Itinerary/ModalSuggestedVias.tsx [hunk 7]
```diff
@@ -110,7 +116,10 @@ const ModalSuggestedVias = ({ suggestedVias }: ModalSuggestedViasProps) => {
           <button
             className="btn btn-danger btn-sm btn-block mb-1"
             type="button"
-            onClick={() => dispatch(clearVias())}
+            onClick={() => {
+              const newPathSteps = [pathSteps.at(0) || null, pathSteps.at(-1) || null];
```
Comment (SharglutDev) [inline front/src/modules/pathfinding/components/Itinerary/ModalSuggestedVias.tsx]: This modal can't exist if we miss origin or destination (l.114 in `Itinerary.tsx`) so we should be able to do this
```suggestion
              const newPathSteps = [pathSteps.at(0)!, pathSteps.at(-1)!;
```

File: front/src/modules/pathfinding/components/Itinerary/ModalSuggestedVias.tsx [hunk 8]
```diff
@@ -32,7 +35,10 @@ const ModalSuggestedVias = ({ suggestedVias }: ModalSuggestedViasProps) => {
     [destination]
   );
 
-  const removeViaFromPath = (op: SuggestedOP) => dispatch(removeVia(op));
+  const removeViaFromPath = (op: SuggestedOP) => {
+    const newPathSteps = pathSteps.filter((step) => !step || matchPathStepAndOp(step, op));
```
Comment (SharglutDev) [inline front/src/modules/pathfinding/components/Itinerary/ModalSuggestedVias.tsx]: I don't see how this could have ever work. Here we filter the steps which matches the op we want to remove so we will have only the removed op in path steps at the end.

Also, same as the other comment, here we know we won't have any null `pathSteps` so we should be able to remove the `!step` (if you did the suggested changed at the top of the file)

```suggestion
    const newPathSteps = pathSteps.filter((step) => !matchPathStepAndOp(step, op));
```

File: front/src/modules/pathfinding/components/Itinerary/ModalSuggestedVias.tsx [hunk 9]
```diff
@@ -9,21 +9,24 @@ import ModalBodySNCF from 'common/BootstrapSNCF/ModalSNCF/ModalBodySNCF';
...
-const ModalSuggestedVias = ({ suggestedVias }: ModalSuggestedViasProps) => {
-  const { upsertViaFromSuggestedOP, clearVias, removeVia } = useOsrdConfActions();
-  const { getVias, getDestination } = useOsrdConfSelectors();
+const ModalSuggestedVias = ({ suggestedVias, launchPathfinding }: ModalSuggestedViasProps) => {
+  const { upsertViaFromSuggestedOP } = useOsrdConfActions();
+  const { getVias, getDestination, getPathSteps } = useOsrdConfSelectors();
   const dispatch = useAppDispatch();
   const vias = useSelector(getVias());
   const destination = useSelector(getDestination);
+  const pathSteps = useSelector(getPathSteps);
```
Comment (SharglutDev) [inline front/src/modules/pathfinding/components/Itinerary/ModalSuggestedVias.tsx]: We could add this here so we don't have to deal with `null` `PathStep` everywhere inside.
```suggestion
  const pathSteps = useSelector(getPathSteps) as PathStep[]; // this modal can't be open if we miss origin or destination
```

File: front/src/modules/pathfinding/components/Itinerary/ModalSuggestedVias.tsx [hunk 10]
```diff
@@ -32,7 +35,10 @@ const ModalSuggestedVias = ({ suggestedVias }: ModalSuggestedViasProps) => {
     [destination]
   );
 
-  const removeViaFromPath = (op: SuggestedOP) => dispatch(removeVia(op));
+  const removeViaFromPath = (op: SuggestedOP) => {
+    const newPathSteps = pathSteps.filter((step) => !step || matchPathStepAndOp(step, op));
+    launchPathfinding(newPathSteps);
```
Comment (SharglutDev) [inline front/src/modules/pathfinding/components/Itinerary/ModalSuggestedVias.tsx:40]: To avoid relaunching the pathfinding if we remove a via which was already in the suggested op list, maybe we could pass to this component `pathProperties.suggestedOperationalPoints` (the pure list of ops on the path) to check if it was in it or not.

> [!NOTE]
> When the PR to "add a op via by map click" will be merged, maybe we will need some day to difference this kind of op from the others

File: front/src/modules/pathfinding/components/Itinerary/DisplayItinerary/Vias.tsx [hunk 11]
```diff
@@ -87,7 +91,10 @@ const Vias = ({ zoomToFeaturePoint, shouldManageStopDuration }: ViasProps) => {
                         data-testid="delete-via-button"
                         className="btn btn-sm btn-only-icon btn-white ml-auto"
                         type="button"
-                        onClick={() => dispatch(deleteVia(index))}
+                        onClick={() => {
+                          const newPathSteps = removeElementAtIndex(pathSteps, index + 1);
+                          launchPathfinding(newPathSteps);
```
Comment (SharglutDev) [inline front/src/modules/pathfinding/components/Itinerary/DisplayItinerary/Vias.tsx:96]: Same suggestion as the one in `ModalSuggestedVias`

File: front/src/modules/pathfinding/components/Pathfinding/Pathfinding.tsx [hunk 12]
```diff
@@ -98,13 +93,13 @@ const Pathfinding = ({ pathProperties, setPathProperties }: PathfindingProps) =>
...
-                  ? t('InvalidTrainScheduleStep')
-                  : t('pathfindingError', { errorMessage: t(pathfindingState.error) })}
+                {pathfindingState.error
+                  ? t('pathfindingError', { errorMessage: t(pathfindingState.error) })
+                  : t('InvalidTrainScheduleStep')}
               </span>
             </div>
           )}
-          {pathfindingState.missingParam && (
+          {pathfindingState.isMissingParam && (
```
Comment (SharglutDev) [inline front/src/modules/pathfinding/components/Pathfinding/Pathfinding.tsx:102]: Strangely, if I remove the origin or destination by clicking on the cross from `Origin` or `Destination` components, this doesn't update this part of the component (probably just an update from the store doesn't rerender). Maybe `isMissingParam` should keep the actual missing param so this could update when removing one of them ?

I think this was broken before but this could be a good moment to fix it.

File: front/src/modules/pathfinding/hooks/usePathfinding.ts [hunk 13]
```diff
@@ -243,116 +86,131 @@ export const usePathfinding = (
...
+      if (infra?.state !== 'CACHED') {
+        return;
+      }
+
+      setIsRunning();
+      const pathfindingInput = getPathfindingQuery({
+        infraId,
+        rollingStock,
+        pathSteps: steps.filter(
+          (step) => step !== null && step.coordinates !== null && !step.isInvalid
```
Comment (SharglutDev) [inline front/src/modules/pathfinding/hooks/usePathfinding.ts]: Why do we need to filter the steps without coordinates for a pathfinding ?

File: front/src/modules/pathfinding/hooks/usePathfinding.ts [hunk 14]
```diff
@@ -243,116 +86,131 @@ export const usePathfinding = (
...
+  };
+
+  const launchPathfinding = useCallback(
+    async (steps: (PathStep | null)[]) => {
+      dispatch(replaceItinerary(steps));
+      setPathProperties(undefined);
 
-    const startPathFinding = async () => {
-      if (pathfindingState.running) {
+      if (steps.some((step) => step === null)) {
```
Comment (SharglutDev) [inline front/src/modules/pathfinding/hooks/usePathfinding.ts:181]: If you replace this line with this, typescript will understand that steps can't have any `null` value and you can avoid all null checks in the rest of the function and in `getPathfindingQuery`

```suggestion
      if (!steps.every((step) => step !== null)) {
```

File: front/src/modules/pathfinding/components/Itinerary/Itinerary.tsx [hunk 15]
```diff
@@ -132,7 +129,12 @@ const Itinerary = ({
               className="col ml-1 my-1 text-white btn bg-info btn-sm"
               type="button"
               onClick={() =>
-                openModal(<ModalSuggestedVias suggestedVias={pathProperties.allWaypoints} />)
+                openModal(
+                  <ModalSuggestedVias
+                    suggestedVias={pathProperties.allWaypoints}
+                    launchPathfinding={launchPathfinding}
```
Comment (SharglutDev) [inline front/src/modules/pathfinding/components/Itinerary/Itinerary.tsx:134]: Can't this component call `launchPathfinding` itself ?

File: front/src/modules/pathfinding/components/Pathfinding/TypeAndPath.tsx [hunk 16]
```diff
@@ -171,8 +170,8 @@ const TypeAndPath = ({ setDisplayTypeAndPath }: TypeAndPathProps) => {
           id: nextId(),
         }));
 
-      dispatch(replaceItinerary(pathSteps));
       setDisplayTypeAndPath(false);
+      launchPathfinding(pathSteps);
```
Comment (SharglutDev) [inline front/src/modules/pathfinding/components/Pathfinding/TypeAndPath.tsx:174]: Seems a little weird that the function where we call this has the same name ()eslint doesn't complain strangely).

**Review by SharglutDev — COMMENTED** [review]
Great refacto, thanks for that !

File: front/src/modules/pathfinding/components/Itinerary/ModalSuggestedVias.tsx (hunk 7 above)
Comment (clarani) [reply front/src/modules/pathfinding/components/Itinerary/ModalSuggestedVias.tsx]: Done

File: front/src/modules/pathfinding/components/Itinerary/ModalSuggestedVias.tsx (hunk 9 above)
Comment (clarani) [reply front/src/modules/pathfinding/components/Itinerary/ModalSuggestedVias.tsx]: You're right, done :+1:

File: front/src/modules/pathfinding/components/Itinerary/ModalSuggestedVias.tsx (hunk 8 above)
Comment (clarani) [reply front/src/modules/pathfinding/components/Itinerary/ModalSuggestedVias.tsx]: Done !

File: front/src/modules/pathfinding/hooks/usePathfinding.ts (hunk 13 above)
Comment (clarani) [reply front/src/modules/pathfinding/hooks/usePathfinding.ts]: We should only filter the invalid steps, you're right :+1:

File: front/src/modules/pathfinding/components/Pathfinding/TypeAndPath.tsx (hunk 16 above)
Comment (clarani) [reply front/src/modules/pathfinding/components/Pathfinding/TypeAndPath.tsx:174]: The trick here is that the function is `launchPathFinding` and not `launchPathfinding` ;)

I can change that but I think it's ok like this too

File: front/src/modules/pathfinding/components/Itinerary/Itinerary.tsx (hunk 15 above)
Comment (clarani) [reply front/src/modules/pathfinding/components/Itinerary/Itinerary.tsx:134]: No I can't because it's a modal, and it is not rendered within the ManageTrainSchedule ContextProvider :/

File: front/src/modules/pathfinding/hooks/usePathfinding.ts (hunk 14 above)
Comment (clarani) [reply front/src/modules/pathfinding/hooks/usePathfinding.ts:181]: Hum I still have the ts error, how did you do ?

File: front/src/modules/pathfinding/components/Pathfinding/Pathfinding.tsx (hunk 12 above)
Comment (clarani) [reply front/src/modules/pathfinding/components/Pathfinding/Pathfinding.tsx:102]: Can we do this in another PR ?

File: front/src/modules/pathfinding/components/Itinerary/ModalSuggestedVias.tsx (hunk 10 above)
Comment (clarani) [reply front/src/modules/pathfinding/components/Itinerary/ModalSuggestedVias.tsx:40]: Hum no, unfortunately, we need to relaunch the pathfinding every time we delete a via, because maybe it modifies the path found by the algorithm.

Here is an example of workflow which will be broken:
- I create a path between A and C
- I add a via B on the map which changes the pathfinding
- the suggested OPs contain now B'
- I add a via on B'
- I remove B, the pathfinding is launched again and is still the same thanks to B'
- I remove B', but since it was added from the suggested OPs, the pathfinding is not relaunched
- however, if I refresh the page, the path will be different (initial path between A and C)

File: front/src/modules/pathfinding/components/Itinerary/DisplayItinerary/Vias.tsx (hunk 11 above)
Comment (clarani) [reply front/src/modules/pathfinding/components/Itinerary/DisplayItinerary/Vias.tsx:96]: I think it does not work, because I can have an origin, a destination and several vias, and then remove the origin, the component will still be displayed :/

File: front/src/applications/operationalStudies/hooks/useManageTrainScheduleContext.tsx [hunk 17]
```diff
@@ -0,0 +1,92 @@
...
+import type { RangedValue } from 'common/types';
+import getPathVoltages from 'modules/pathfinding/helpers/getPathVoltages';
+import usePathfinding from 'modules/pathfinding/hooks/usePathfinding';
+import type { PathfindingState } from 'modules/pathfinding/types';
+import { upsertPathStepsInOPs } from 'modules/pathfinding/utils';
+import type { PathStep } from 'reducers/osrdconf/types';
+
+import type { ManageTrainSchedulePathProperties } from '../types';
+
+type ManageTrainScheduleContextType = {
```
Comment (anisometropie) [inline front/src/applications/operationalStudies/hooks/useManageTrainScheduleContext.tsx:12]: why do we carry that `manage` every where. can’t this just be TrainScheduleContextType ?

File: front/src/applications/operationalStudies/hooks/useManageTrainScheduleContext.tsx [hunk 18]
```diff
@@ -0,0 +1,92 @@
...
+}: ManageTrainScheduleContextProviderProps) => {
+  const { getPathSteps } = useOsrdConfSelectors();
+  const pathSteps = useSelector(getPathSteps);
+
+  const [pathProperties, setPathProperties] = useState<ManageTrainSchedulePathProperties>();
+
+  const { launchPathfinding, pathfindingState, infraInfo } = usePathfinding(setPathProperties);
+
+  useEffect(() => {
+    if (pathProperties) {
```
Comment (anisometropie) [inline front/src/applications/operationalStudies/hooks/useManageTrainScheduleContext.tsx]: the dependencies array is empty, so if this useEffect is called only once on the first render , why would pathProperties not be undefined?

File: front/src/applications/operationalStudies/hooks/useManageTrainScheduleContext.tsx [hunk 19]
```diff
@@ -0,0 +1,92 @@
...
+
+export const ManageTrainScheduleContextProvider = ({
+  children,
+}: ManageTrainScheduleContextProviderProps) => {
+  const { getPathSteps } = useOsrdConfSelectors();
+  const pathSteps = useSelector(getPathSteps);
+
+  const [pathProperties, setPathProperties] = useState<ManageTrainSchedulePathProperties>();
+
+  const { launchPathfinding, pathfindingState, infraInfo } = usePathfinding(setPathProperties);
```
Comment (anisometropie) [inline front/src/applications/operationalStudies/hooks/useManageTrainScheduleContext.tsx:30]: I kinda fail to see what more that context does than just using `usePathfinding` directly.

File: front/src/modules/pathfinding/components/Itinerary/ModalSuggestedVias.tsx (hunk 10 above)
Comment (SharglutDev) [reply front/src/modules/pathfinding/components/Itinerary/ModalSuggestedVias.tsx:40]: You're totally right !

File: front/src/modules/pathfinding/components/Pathfinding/Pathfinding.tsx (hunk 12 above)
Comment (anisometropie) [reply front/src/modules/pathfinding/components/Pathfinding/Pathfinding.tsx:102]: yes, keep this separate

Comment (SharglutDev) [reply front/src/modules/pathfinding/components/Pathfinding/Pathfinding.tsx:102]: Sure I created an issue

File: front/src/modules/pathfinding/components/Pathfinding/TypeAndPath.tsx (hunk 16 above)
Comment (anisometropie) [reply front/src/modules/pathfinding/components/Pathfinding/TypeAndPath.tsx:174]: This can be really confusing to see which one is called. I would call that `handleClick` since it’s an event handler

File: front/src/modules/trainschedule/components/ManageTrainSchedule/ManageTrainScheduleMap/setPointIti.ts [hunk 20]
```diff
@@ -1,35 +1,37 @@
...
-      store.dispatch(updateDestination(pathStep));
+      newPathSteps = replaceElementAtIndex(pathSteps, -1, pathStep);
       break;
     default:
       if (pathProperties) {
-        store.dispatch(addVia({ newVia: pathStep, pathProperties }));
+        newPathSteps = insertViaFromMap(pathSteps, pathStep, pathProperties);
       } else {
-        store.dispatch(
-          replaceItinerary(addElementAtIndex(pathSteps, pathSteps.length - 1, pathStep))
```
Comment (anisometropie) [inline front/src/modules/trainschedule/components/ManageTrainSchedule/ManageTrainScheduleMap/setPointIti.ts:30]: how is this updated in the store now?

**Review by anisometropie — CHANGES_REQUESTED** [review]

File: front/src/modules/pathfinding/hooks/usePathfinding.ts (hunk 14 above)
Comment (SharglutDev) [reply front/src/modules/pathfinding/hooks/usePathfinding.ts:181]: hmmm 

Capture d’écran 2024-12-12 à 17 31 05

File: front/src/modules/pathfinding/components/Itinerary/Itinerary.tsx (hunk 15 above)
Comment (SharglutDev) [reply front/src/modules/pathfinding/components/Itinerary/Itinerary.tsx:134]: Oh right ok

File: front/src/modules/pathfinding/components/Pathfinding/TypeAndPath.tsx (hunk 16 above)
Comment (SharglutDev) [reply front/src/modules/pathfinding/components/Pathfinding/TypeAndPath.tsx:174]: Haha ok, took me a while to spot the difference. Maybe it's still better to rename it

File: front/src/modules/pathfinding/components/Itinerary/DisplayItinerary/Vias.tsx (hunk 11 above)
Comment (SharglutDev) [reply front/src/modules/pathfinding/components/Itinerary/DisplayItinerary/Vias.tsx:96]: Right, my mistake. I thought this would remove it as well.

File: front/src/modules/pathfinding/hooks/usePathfinding.ts (hunk 13 above)
Comment (SharglutDev) [reply front/src/modules/pathfinding/hooks/usePathfinding.ts]: With your check l.181, we can be sure that all path steps aren't null so it should be ok to remove the `=== null` as well

File: front/src/applications/operationalStudies/hooks/useManageTrainScheduleContext.tsx (hunk 17 above)
Comment (clarani) [reply front/src/applications/operationalStudies/hooks/useManageTrainScheduleContext.tsx:12]: Hum I would rather keep the name of the view, even if I agree it's not the best naming ever. If you want to rename it, we can do it in another PR

File: front/src/modules/trainschedule/components/ManageTrainSchedule/ManageTrainScheduleMap/setPointIti.ts (hunk 20 above)
Comment (clarani) [reply front/src/modules/trainschedule/components/ManageTrainSchedule/ManageTrainScheduleMap/setPointIti.ts:30]: The store is updated in launchPathfinding

File: front/src/applications/operationalStudies/hooks/useManageTrainScheduleContext.tsx (hunk 19 above)
Comment (clarani) [reply front/src/applications/operationalStudies/hooks/useManageTrainScheduleContext.tsx:30]: As we discussed it yesterday, I think we can resolve this conversation and talk about it next week ?

File: front/src/applications/operationalStudies/hooks/useManageTrainScheduleContext.tsx (hunk 18 above)
Comment (clarani) [reply front/src/applications/operationalStudies/hooks/useManageTrainScheduleContext.tsx]: I think I need to check with my other PR, but the dependencies should not be empty

File: front/src/modules/pathfinding/components/Pathfinding/TypeAndPath.tsx (hunk 16 above)
Comment (clarani) [reply front/src/modules/pathfinding/components/Pathfinding/TypeAndPath.tsx:174]: Done

File: front/src/modules/pathfinding/hooks/usePathfinding.ts (hunk 14 above)
Comment (SharglutDev) [reply front/src/modules/pathfinding/hooks/usePathfinding.ts:181]: Ok so looks like I have the typescript 5.5+ on my setup (maybe because of vscode) which fixes this behavior but the app is still in 5.3.

We should keep this in mind when bumping typescript version

File: front/src/applications/operationalStudies/hooks/useManageTrainScheduleContext.tsx (hunk 18 above)
Comment (anisometropie) [reply front/src/applications/operationalStudies/hooks/useManageTrainScheduleContext.tsx]: it is empty line 51, so it’s only called once. and pathProperties has no reason to be defined here. so it looks like this useEffect never does anything

File: front/src/modules/pathfinding/hooks/usePathfinding.ts (hunk 13 above)
Comment (SharglutDev) [reply front/src/modules/pathfinding/hooks/usePathfinding.ts]: Same as next comment. Added issue here : https://github.com/OpenRailAssociation/osrd/issues/10081

File: front/src/modules/pathfinding/hooks/usePathfinding.ts (hunk 14 above)
Comment (SharglutDev) [reply front/src/modules/pathfinding/hooks/usePathfinding.ts:181]: issue is here : https://github.com/OpenRailAssociation/osrd/issues/10081

**Review by SharglutDev — APPROVED** [review]
Lgtm and tested, great job !

File: front/src/applications/operationalStudies/hooks/useManageTrainScheduleContext.tsx (hunk 18 above)
Comment (clarani) [reply front/src/applications/operationalStudies/hooks/useManageTrainScheduleContext.tsx]: Ok, I can remove it for now :+1:

**Review by anisometropie — APPROVED** [review]
thanks, good reworking

**Review by emersion — APPROVED** [review]

**Review by SharglutDev — APPROVED** [review]
Lgtm and tested again :)
